client/src/components/Grid/GridElements/GridLink.vue

```diff
@@ -0,0 +1,10 @@
+<script setup lang="ts">
+interface Props {
+    text: string;
+}
+defineProps<Props>();
+</script>
+
+<template>
+    <a class="font-weight-bold" href="#" @click.prevent="$emit('click')">{{ text }}</a>
```

Comment:
In Typescript components, explicitly declare emits as follows:

```vue
<script setup lang="ts">
const emit = defineEmits<{
    (e: "click"): void;
}>();
</script>

<template>
    <button @click="emit('click')" />
</template>
```

This ensures type-safety on both the consuming component and the emitting component, should the events function signature change. `$emit` is not type-safe